tests/framework/extensions/rke1/nodepools/nodepools.go

```diff
@@ -59,14 +60,14 @@ func NodePoolSetup(client *rancher.Client, nodeRoles []NodeRoles, ClusterID, Nod
 
 // ScaleWorkerNodePool is a helper method that will add a worker node pool to the existing RKE1 cluster. Once done, it will scale
 // the worker node pool to add a worker node, scale it back down to remove the worker node, and then delete the worker node pool.
-func ScaleWorkerNodePool(client *rancher.Client, nodeRoles []NodeRoles, ClusterID, NodeTemplateID string) error {
+func ScaleNewWorkerNodePool(client *rancher.Client, cluster *management.Cluster, nodeTemplate *nodetemplates.NodeTemplate) error {
```

Comment:
I think it would make sense to update this function to match what you've added for rke2/k3s where the quantity is specified by the user. 

This function is currently doing what I would consider part of the 'test' instead of the work for the test. Since there are hard coded values here, this can't really be used for anything other than the test it was written for, in regard to scaling.

Reply:
100% agree with this. This extension needs to be reusable adding "test" validations in here should be avoided.

Reply:
This is an excellent point. I reworked this to match RKE2/K3S and it looks way better now